src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -0,0 +1,600 @@
+        Persistence.otpUsers.removeById(user.id);
+        for (MonitoredTrip trip : Persistence.monitoredTrips.getFiltered(eq("userId", user.id))) {
+            PersistenceTestUtils.deleteMonitoredTrip(trip);
+        }
+    }
+
+    @AfterEach
+    public void tearDownAfterTest() {
+        OtpTestUtils.resetOtpMocks();
+        DateTimeUtils.useSystemDefaultClockAndTimezone();
+    }
+
+    /**
+     * To run this trip, change the env.yml config values for OTP_API_ROOT
+     * (and OTP_PLAN_ENDPOINT) to a valid OTP server.
+     */
+    @Test
+    public void canMonitorTrip() throws URISyntaxException, CloneNotSupportedException {
+        // Do not run this test on Travis CI because it requires a live OTP server
+        // FIXME: Add live otp server to e2e tests.
+        assumeTrue(!ConfigUtils.isRunningCi && CommonTestUtils.IS_END_TO_END);
+        MonitoredTrip monitoredTrip = new MonitoredTrip(OtpTestUtils.sendSamplePlanRequest());
+        monitoredTrip.updateAllDaysOfWeek(true);
+        monitoredTrip.userId = user.id;
+        monitoredTrip.tripName = "My Morning Commute";
+        monitoredTrip.itineraryExistence = new ItineraryExistence();
+        monitoredTrip.itineraryExistence.monday = new ItineraryExistence.ItineraryExistenceResult();
+        Persistence.monitoredTrips.create(monitoredTrip);
+        LOG.info("Created trip {}", monitoredTrip.id);
+
+        // Setup an OTP mock response in order to trigger some of the monitor checks.
+        Itinerary mockMondayJune15Itinerary = OtpTestUtils.OTP_DISPATCHER_PLAN_RESPONSE
+            .getResponse().plan.itineraries.get(0);
+
+        // parse original itinerary date/time and then update mock itinerary to occur on Monday June 15
+        OtpTestUtils.updateBaseItineraryTime(
+            mockMondayJune15Itinerary,
+            DateTimeUtils.makeOtpZonedDateTime(mockMondayJune15Itinerary.startTime)
+                .withDayOfMonth(15)
+        );
+
+        // Add fake alerts to simulated itinerary.
+        ArrayList<LocalizedAlert> fakeAlerts = new ArrayList<>();
+        fakeAlerts.add(new LocalizedAlert());
+        mockMondayJune15Itinerary.legs.get(1).alerts = fakeAlerts;
+
+        OtpTestUtils.setupOtpMocks(List.of(OtpTestUtils.OTP_DISPATCHER_PLAN_RESPONSE.getResponse()));
+
+        // mock the current time to be 8:45am on Monday, June 15
+        DateTimeUtils.useFixedClockAt(
+            noonMonday8June2020
+                .withDayOfMonth(15)
+                .withHour(8)
+                .withMinute(45)
+        );
+
+        // Next, run a monitor trip check from the new monitored trip using the simulated response.
+        CheckMonitoredTrip checkMonitoredTrip = new CheckMonitoredTrip(monitoredTrip);
+        checkMonitoredTrip.run();
+        // Assert that there is one notification generated during check.
+        // TODO: Improve assertions to use snapshots.
+        Assertions.assertEquals(1, checkMonitoredTrip.notifications.size());
```

Comment:
canMonitorTrip() test fails:

org.opentest4j.AssertionFailedError: 
Expected :1
Actual   :3

Reply:
Bug introduced in 1987213b6754ffa98f4748820b628be64f9443f4. Fixed in d95910c71758df3212e5f878a91a509f02839dd0.